File: java/2487. Remove Nodes From Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNodes(ListNode head) {
        ListNode reversed = reverse(head);
        ListNode pointer = reversed;
        while (pointer != null) {
            while (pointer.next != null && pointer.next.val < pointer.val) {
                pointer.next = pointer.next.next;
            }
            pointer = pointer.next;
        }
        return reverse(reversed);
    }
    private ListNode reverse(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode reversed = reverse(head.next);
        head.next.next = head;
        head.next = null;
        return reversed;
    }
}

File: python/2487. Remove Nodes From Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        reversed = self.reverse(head)
        pointer = reversed
        while pointer != None:
            while pointer.next != None and pointer.next.val < pointer.val:
                pointer.next = pointer.next.next
            pointer = pointer.next
        return self.reverse(reversed)
    def reverse(self, head):
        previous = None
        while head != None:
            temp = head.next
            head.next = previous
            previous = head
            head = temp
        return previous